src/tutorial/developing-more-applicative-plugin.md

```diff
@@ -0,0 +1,105 @@
+# Developing More Applicative Plugin
+
+Now you have learned the basics of developing Denpos plugins in the previous sections.
+Then it would be best if you tried to create a more functional plugin.
+
+So let me ask you, out of the blue, have you ever itched to solve mazes while programming?
+I never have.
+In any case, there may be people who love solving mazes and can't get enough of it.
+So let's try to develop a Denops plugin that can generate and display a maze in vim at any time.
+
+Of course, it would be nice to start by coding a maze generation algorithm.
+However, you are now with Deno so that you can use a third-party library [maze_generator](https://deno.land/x/maze_generator@v0.4.0) for your convenience.
+First, you should define a `Maze` command similarly to `HelloWorldEcho`; `Maze` generates a maze and outputs it with `console.log()`.
+
+```ts:main.ts
+import { Denops } from "https://deno.land/x/denops_std@v1.0.0-beta.2/mod.ts";
+import { Maze } from "https://deno.land/x/maze_generator@v0.4.0/mod.js";
+
+export async function main(denops: Denops): Promise<void> {
+  denops.dispatcher = {
+    async maze(): Promise<void> {
+      const maze = new Maze({}).generate();
+      const content = maze.getString();
+      console.log(content);
+    },
+  };
+
+  await denops.cmd(`command! Maze call denops#request('${denops.name}', 'maze', [])`);
+};
+```
+
+Restarting Vim, and you will see a maze by commands:
+
+```vim
+:Maze
+:mes
+```
+
+![](../img/developing-more-applicative-plugin-1.png)
+
+Well done! But it is a little boring... So let's try to modify the code to make the maze output to a buffer.
+
+```ts:main.ts
+import { Denops } from "https://deno.land/x/denops_std@v1.0.0-beta.2/mod.ts";
+import { Maze } from "https://deno.land/x/maze_generator@v0.4.0/mod.js";
+
+export async function main(denops: Denops): Promise<void> {
+  denops.dispatcher = {
+    async maze(): Promise<void> {
+      const maze = new Maze({}).generate();
+      const content = maze.getString();
+      await denops.cmd("enew");
+      await denops.call("setline", 1, content.split(/\n/));
+    },
+  };
+
+  await denops.cmd(`command! Maze call denops#request('${denops.name}', 'maze', [])`);
+};
+```
+
+In this code, `denops.cmd()` calls the `enew` command of vim to open a new buffer in the current window and then `denops.call()` calls `setline()` to write the maze to the buffer.
```

Comment:
Suggestion

How about this translation?

> ... and then `denops.call()` calls the `setline()` function to write the maze ...

This contrast makes it clearer that `denops.cmd()` executes Vim *commands*, while `denops.call()` calls Vim *functions*.

Reply:
Thank you for your review and I agree with your intension to clearify difference of the meanings between Vim _commands_ and Vim _functions_.
So I've revised as in 882ddc9.